### PR TITLE
Improve agg_p_nom_limits configuration

### PR DESCRIPTION
**Pull request by tgi-climact**
## Changes proposed in this Pull Request

Improved the behaviour of `agg_p_nom_limits`:

  - Moved the associated configuration to `solving`. This allows *Snakemake* to correctly decide which rules to run when the configuration changes.

  - Added the ability to enable aggregation of all *offwind* types (*offwind-ac* and *offwind-dc*) when writing the constraint.

  - Added the possibility to take existing capacities into account when writing the constraint.

  - Added the possibility to have a different file for each planning horizon.

## Checklist

- [x] I tested my contribution locally and it seems to work fine.
- [x] Code and workflow changes are sufficiently documented.
- [ ] Changed dependencies are added to `envs/environment.yaml`.
- [x] Changes in configuration options are added in all of `config.default.yaml`.
- [x] Changes in configuration options are also documented in `doc/configtables/*.csv`.
- [x] A release note `doc/release_notes.rst` is added.

### CODE REVIEW

File: config/config.default.yaml [hunk 1]
```diff
@@ -758,6 +757,14 @@ solving:
     linearized_unit_commitment: true
     horizon: 365
 
+  agg_p_nom_limits:
+    agg_offwind: false
+    include_existing: false
+    years:
+      2030: data/agg_p_nom_minmax.csv
+      2040: data/agg_p_nom_minmax.csv
+      2050: data/agg_p_nom_minmax.csv
```
Comment (FabianHofmann) [inline config/config.default.yaml:766]: thanks for the contribution. I am but I see that the functionality is needed. perhaps we can add columns to the data file for different years instead?

Comment (tgi-climact) [reply config/config.default.yaml]: @FabianHofmann I've pushed an update.

**Comment by tgi-climact** [comment]
@FabianHofmann Would it be useful to add some code to extend this feature to links ? e.g. write constraint into `agg_p_nom_minmax.csv`for nuclear

**Review by fneum — APPROVED** [review]

**Comment by fneum** [comment]
@tgi-climact all looks good!

Yes, I think it makes sense to extend this feature to links. I'll merge this PR first, and then we can extend.

**Comment by tgi-climact** [comment]
> @tgi-climact all looks good!
> 
> Yes, I think it makes sense to extend this feature to links. I'll merge this PR first, and then we can extend.

Great, I will propose a new PR when I have some time.

**Pull request merged**